src/Servers/Kestrel/Core/src/Internal/Http2/Http2Stream.FeatureCollection.cs

```diff
@@ -73,5 +75,11 @@ async Task IHttpResponseCompletionFeature.CompleteAsync()
                 await ProduceEnd();
             }
         }
+
+        void IHttpResetFeature.Reset(int errorCode)
+        {
+            var abortReason = new ConnectionAbortedException(CoreStrings.FormatHttp2StreamResetByApplication((Http2ErrorCode)errorCode));
```

Comment:
What happens if Reset is called multiple times for a request?

What happens if Reset is called after the request has already been aborted? (e.g. the client sends RST_STREAM)

Reply:
ResetAndAbort is guarded against multiple invocations.
https://github.com/aspnet/AspNetCore/blob/9c4faf86e5534dba490dc0e2f634ae291aaa2cfd/src/Servers/Kestrel/Core/src/Internal/Http2/Http2Stream.cs#L452-L460